Makefile

```diff
@@ -0,0 +1,94 @@
+SHELL := /bin/bash
+
+PACKER_ZIP ?= https://releases.hashicorp.com/packer/1.3.2/packer_1.3.2_linux_amd64.zip
+#PACKER_LOG ?= '1'
+PACKER_LOG ?= '0'
+PACKER_NO_COLOR ?= '1'
+CHECKPOINT_DISABLE ?= '1'
+SPEL_CI ?= false
+SPEL_BUILDERS ?= minimal-rhel-7-hvm,minimal-centos-7-hvm
+SPEL_DESC_URL ?= https://github.com/plus3it/spel
+SPEL_AMIGEN7SOURCE ?= https://github.com/plus3it/AMIgen7.git
+SPEL_AMIUTILSOURCE ?= https://github.com/ferricoxide/Lx-GetAMI-Utils.git
+SPEL_AWSCLISOURCE ?= https://s3.amazonaws.com/aws-cli/awscli-bundle.zip
+SPEL_EPEL7RELEASE ?= https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
+SPEL_CUSTOMREPORPM7 ?= https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
+SPEL_EPELREPO ?= epel
+SPEL_EXTRARPMS ?= https://s3.amazonaws.com/ec2-downloads-windows/SSMAgent/latest/linux_amd64/amazon-ssm-agent.rpm,python36
+SOURCE_AMI_CENTOS7_HVM ?= ami-090b9dabe1c9f40b3
+SOURCE_AMI_RHEL7_HVM ?= ami-0394fe9914b475c53
+SSH_INTERFACE ?= public_dns
+PIP_URL ?= https://bootstrap.pypa.io/get-pip.py
+PYPI_URL ?= https://pypi.org/simple
+
+.PHONY: all install pre_build build post_build
+.EXPORT_ALL_VARIABLES:
+
+#ifeq ("${SPEL_CI}", "true")
+#SPEL_VERSION=$(shell date +%Y.%m.dev%s)
+#endif
+
+$(info SPEL_IDENTIFIER=${SPEL_IDENTIFIER})
+$(info SPEL_VERSION=${SPEL_VERSION})
+
+ifndef SPEL_IDENTIFIER
+$(error SPEL_IDENTIFIER is not set)
+endif
+
+ifndef SPEL_VERSION
+$(error SPEL_VERSION is not set)
+endif
+
+$(shell mkdir -p ".spel/${SPEL_VERSION}/")
+$(shell touch ".spel/${SPEL_VERSION}/packer.log")
+#PACKER_LOG_PATH=".spel/${SPEL_VERSION}/packer.log"
+
+# Check if $SPEL_PIN_AWSCLI_BUNDLE is not null
+ifdef SPEL_PIN_AWSCLI_BUNDLE
+AWSCLI_PIN=$(shell grep aws-cli requirements/aws-cli.txt)
+ifndef AWSCLI_PIN
+$(error AWSCLI_PIN is not set: ${.SHELLSTATUS})
+endif
+
+AWSCLI_VERSION:=$(subst aws-cli==,,${AWSCLI_PIN})
+SPEL_AWSCLISOURCE := $(shell dirname ${SPEL_AWSCLISOURCE}/awscli-bundle-${AWSCLI_VERSION}.zip)
+ifndef SPEL_AWSCLISOURCE
+$(error SPEL_AWSCLISOURCE is not set: ${SHELLSTATUS})
+endif
+endif
+$(info SPEL_AWSCLISOURCE=${SPEL_AWSCLISOURCE})
+
+# Check if $SPEL_SSM_ACCESS_KEY is not null
+ifdef SPEL_SSM_ACCESS_KEY
+SSM_ACCESS_KEY = $(shell aws ssm get-parameters --name ${SPEL_SSM_ACCESS_KEY} --with-decryption --query 'Parameters[0].Value' --out text)
```

Comment:
Do we know what happens when these `shell` invocations fail? Does that cause `make` to fail immediately, or does it ignore the failure and keep going, failing later at a less intuitive spot?

Reply:
Tested this out a bit and it will fail "silently". 

1. Tested putting in an inaccurate AKID and it just returns None and keeps on going.  
2. Testing putting in the wrong command and it prints out the error but ultimately keeps on going

I will add a conditional to check for "None" and have it error out